src/routes/api/api-router.ts

```diff
@@ -20,6 +20,37 @@ ApiRouter.use(urlencoded({ extended: false }));
 ApiRouter.use(json());
 ApiRouter.use(LogMiddleware);
 
+// Verify SLAuth headers to make sure that no open access was allowed for these endpoints
+// And also log how the request was authenticated
+
+ApiRouter.use(
+	async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+		const mechanism = req.get("X-Slauth-Mechanism");
+		const issuer = req.get("X-Slauth-Issuer");
+		const principal = req.get("X-Slauth-Principal");
+
+		req.log = req.log.child({slauth: {
+			mechanism,
+			issuer,
+			principal,
+			userGroup: req.get("X-Slauth-User-Groups"),
+			aaid: req.get("X-Slauth-User-Aaid"),
+			username: req.get("X-Slauth-User-Username")
+		}});
+
+		if(!mechanism || mechanism === "open") {
+			req.log.warn("Attempt to access Admin API without authentication")
+			res.status(401).json({error: "Open access not allowed"});
+			return;
+		}
```

Comment:
Still need to fix this conditional.  You can't base your security on what not to expect.

Reply:
Don't understand why. 
We are checking that headers present, we are not repeating POCO/SLAuth logic here. That's how it works in DSS and any other services which use SLAuth.

Our goal of checking this header to make sure that the SP is enabled (and that the endpoint is not public in the config). That's it.

Please refer to SLAuth docs go/slauth to get better understanding on how it works.